### ISSUE TITLE
How to use Gmail ?

### ISSUE THREAD

**Issue by pscott-au** [sdondley/moo-google#18]
I am able to access the top level API exposed methods such as 

`print Dumper $gapi_agent->Gmail->Users->getProfile( { userId => 'me'  } )->json if ( );`

But I am unable to work out how to use any API method that is more deeply described such as Messages::list etc - are these supposed to work or do we need to access the `$gapi_agent->api_query` lower level method directly for essentially all useful usage?

**Comment by sdondley** [comment]
Yeah, use the api_query method. I was fiddling about with all this about a month ago and I was able to pull off batch calls and other stuff with it. My memory is a bit fuzzy and the code is a giant hack job but it worked. I'll post it to a github repo so you can look at it. I have to check to make sure no credentials are in there before I do that.

**Comment by sdondley** [comment]
OK, I shared a repo with you. It has a module I was using to screw around with Moo::Google. Not meant for release. Look in the "hacked_moose_modules" directory. You'll need that module to get the code in my modules to work properly.

I don't know if I left the code in a working state or not. You may need to fuss with it a bit.

**Comment by pscott-au** [comment]
I have no problem using the Google GMAIL API - the issue is that
WebService::Google::Client doesn't work to access the GMAIL methods deeper
than the 3 top level which makes it of no use and requires reverting to
lower level calls which means that is just as easy to not use the module at
all.

**Comment by sdondley** [comment]
Right. Moo::Google doesn't work with all API calls. You have to use api_query.

**Comment by pscott-au** [comment]
unless your hacked earlier code version from this repo does actually
correctly iterate down the API methods in the discovery document which is
really the key issue - is that what you were doing?

**Comment by sdondley** [comment]
That's what I was planning on doing, yes. I needed to learn how the api_query worked though. So I did some experimenting with it. I did write a crude function to convert an api call that didn't work into one that did.

**Comment by pscott-au** [comment]
lol - ok - never mind - sorry I shoudl just let you do your thing as
probably seems like I'm trolling you - am not - just trying to work out
whether there is anything I can align with .. thanks for the response

**Comment by sdondley** [comment]
Look in the API.pm module in my repo, the "get" method. I have to retrace what I did but it basically enables you to do a get api call with a resource, which you can't do with Moo::Google.

**Comment by sdondley** [comment]
That module also has a "batch" function which was a real bear to figure out. Had to do a lot of hacking to the Client.pm module to get that working.
